lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -450,6 +466,54 @@ function validatePolicyResource(bucketName, policy) {
     });
 }
 
+function checkIp(value) {
+    const isValid = /^(\d{1,3}\.){3}\d{1,3}(\/\d{1,2})?$/.test(value);
+    if (isValid) {
+        return null;
+    }
+    return 'Invalid IP address in Conditions';
+}
+
+// This function checks all bucket policy conditions if the values provided
+// are valid for the condition type. If not it returns a relevant Malformed policy error string
+function validatePolicyConditions(policy) {
+    const validConditions = [
+        { conditionKey: 'aws:SourceIp', conditionValueTypeChecker: checkIp },
+        { conditionKey: 's3:object-lock-remaining-retention-days' },
+    ];
+    // keys where value type does not seem to be checked by AWS:
+    // - s3:object-lock-remaining-retention-days
+
+    // there can be multiple statements in the policy, each with a Condition enclosure
+    for (let i = 0; i < policy.Statement.length; i++) {
+        const s = policy.Statement[i];
+        if (s.Condition) {
+            const conditionOperators = Object.keys(s.Condition);
+            // there can be multiple condition operations in the Condition enclosure
+            for (const conditionOperator of conditionOperators) {
+                const conditionKey = Object.keys(s.Condition[conditionOperator])[0];
+                const conditionValue = s.Condition[conditionOperator][conditionKey];
+                const validCondition = validConditions.find(validCondition =>
+                    validCondition.conditionKey === conditionKey
+                );
+                // AWS returns does not return an error if the condition starts with 'aws:'
+                // so we reproduce this behaviour
```

Comment:
While standard, is it something we want to support, as this will have no effect?
I am thinking about very long condition keys; as the BP is stored in the bucket MD, we want to avoid using all the available space for th BP (yes, this is to protect against some attacks, not the nominal use case :slightly_smiling_face: )

Reply:
There is a conversation to be had here indeed. I have simply been basing myself on AWS behaviour. I'll address it with the team somehow.

Reply:
Ticket created
https://scality.slack.com/archives/C01M77VU8RJ/p1704819662752699?thread_ts=1704792330.505129&cid=C01M77VU8RJ